### PR TITLE
Don't suggest variables explicitly marked unused

### PR DESCRIPTION
**Pull request by drvink**
This is the first half of fixing #3705. The next half is to fix the scope leak itself.

### CODE REVIEW

File: tests/fsharpqa/Source/CompilerOptions/fsc/dumpAllCommandLineOptions/dummy.fsx [hunk 1]
```diff
@@ -95,6 +95,7 @@
 //<Expects status="success">section='- ADVANCED -             ' ! option=readline                       kind=OptionSwitch</Expects>
 //<Expects status="success">section='- ADVANCED -             ' ! option=quotations-debug               kind=OptionSwitch</Expects>
 //<Expects status="success">section='- ADVANCED -             ' ! option=shadowcopyreferences           kind=OptionSwitch</Expects>
+//<Expects status="success">section='- ADVANCED -             ' ! option=simpleresolution               kind=OptionUnit</Expects>
```
Comment (forki) [inline tests/fsharpqa/Source/CompilerOptions/fsc/dumpAllCommandLineOptions/dummy.fsx]: ;-) ouch

File: src/fsharp/ErrorResolutionHints.fs [hunk 2]
```diff
@@ -43,7 +43,11 @@ let FilterPredictions (idText:string) (suggestionF:ErrorLogger.Suggestions) =
     if allSuggestions.Contains idText then [] else // some other parsing error occurred
     allSuggestions
     |> Seq.choose (fun suggestion ->
-        if IsOperatorName suggestion then None else
+        // Because beginning a name with _ is used both to indicate an unused
+        // value as well as to formally squelch the associated compiler
+        // error/warning (FS1182), we remove such names from the suggestions,
+        // both to prevent accidental usages as well as to encourage good taste
```
Comment (forki) [inline src/fsharp/ErrorResolutionHints.fs]: Thanks for making this explicit!

**Comment by drvink** [comment]
May as well just merge this one now, since it doesn't impact work on #3710 or anything else, right? If the other ticket somehow falls by the wayside all the way until another release, at least we won't be giving the silly suggestion.

**Comment by forki** [comment]
This one is absolutely fine to go in.

Am 11.10.2017 19:12 schrieb "Mark Laws" <notifications@github.com>:

**Comment by dsyme** [comment]
Thanks!

**Pull request merged**